/**:
  ros__parameters:

    # frame id
    frame_id_slam_frame: "slam"



    # input
    topic_sub_odom:  "fastlio2/odometry"               # NOTE: SLAM to sensor frame
    topic_sub_cloud: "fastlio2/cloud/registered_body"  # NOTE:      in sensor frame



    # keyframe selection
    keyframe_gap_meter:  1.5
    keyframe_gap_deg:   15.0



    # voxel grid filter
    leaf_size_keyframe: 0.2
    leaf_size_icp:      0.2
    leaf_size_out_map:  0.1  # NOTE: for output, "pgo/out/map"



    # loop candidate
    lc_max_radius_m:     3.0
    lc_min_time_diff_s: 20.0



    # ICP test
    icp_type:                  1     # NOTE: ICP (0),  G-ICP (1)
    icp_stack_size:           20     # NOTE: kf_clouds[prv - this ~ prv + this] make ICP target
    icp_config_max_iter:     100     # NOTE: max. iteration
    icp_config_max_cor_dist:  10.0   # NOTE: max. corresponding distance in [m]
    icp_test_max_fitness:      0.04



    # GTSAM factor noise variance
    # NOTE: variance = (max_noise / 3) * (max_noise / 3)
    var_prior: 1e-10
    var_odom:  0.001
    var_loop:  0.01



    # GTSAM pose graph optimization
    num_update: 5



    # output
    # NOTE: pgo/out/map
    enable_out_map: true  # FIX: use param



    # visualization
    # NOTE: pgo/keyframe/cloud
    enable_pub_keyframe_cloud: true

    # NOTE: scpgo/graph
    enable_pub_graph: true  # FIX: use param
    nodes_scale:      0.15
    nodes_alpha:      0.5
    edges_scale:      0.05
    edges_alpha:      0.5
    loops_pass_scale: 0.05
    loops_pass_alpha: 0.5
    loops_fail_scale: 0.05
    loops_fail_alpha: 0.5

    # NOTE: pgo/icp/source
    # NOTE: pgo/icp/target
    # NOTE: pgo/icp/aligned
    enable_pub_icp: true
